use-cases/xtclim/train.py

```diff
@@ -7,24 +7,31 @@
 import argparse
 import logging
 from datetime import datetime
+import yaml
 
 sys.path.append(os.path.join(os.path.dirname(__file__), 'src'))
 sys.path.append(os.path.join(os.path.dirname(__file__), 'preprocessing'))
 
-from itwinai.parser import ConfigParser, ArgumentParser
+from itwinai.parser import ConfigParser
 
-if __name__ == "__main__":
-    parser = ArgumentParser()
-    parser.add_argument(
-         "-p", "--pipeline", type=str, required=True,
-         help='Configuration file to the pipeline to execute.'
-    )
-    args = parser.parse_args()
+def read_config(file_path):
+    with open(file_path, 'r') as f:
+        config = yaml.safe_load(f)
+    return config
 
-    pipe_parser = ConfigParser(
-         config=args.pipeline,
-    )
+if __name__ == "__main__":
 
-    pipeline = pipe_parser.parse_pipeline()
-    pipeline.execute()
+    config = read_config('pipeline.yaml')
+    seasons_list = config['seasons']
+    for season in seasons_list:
+        config['pipeline']['init_args']['steps']['training-step']['init_args']['seasons'] = season
+        model_uri = f"outputs/cvae_model_{season}1d_1memb.pth"
+        config['pipeline']['init_args']['steps']['evaluation-step']['init_args']['model_uri'] = model_uri
+        config['pipeline']['init_args']['steps']['evaluation-step']['init_args']['seasons'] = season
+        pipe_parser = ConfigParser(
+            config=config,
+        )
+        pipeline = pipe_parser.parse_pipeline()
+        print(f"Running pipeline for season: {season}")
+        pipeline.execute()
```

Comment:
the comment in hpo.py on breaking down pipeline into preprocessing steps (potentially always the same across seasons) and training steps, to avoid re-doing the preprocessing. However, I am not an expert of this use case and this may not be feasible for some reason